resources/scenic-analysis.wrattler

```diff
@@ -0,0 +1,157 @@
+
+# (for now) remove columns/dates that have nan values
+dates_df = crime_data[list(crime_data.columns)[3:]]
+nans = lambda df: df[df.isnull().any(axis=1)]
+nan_df = nans(dates_df)
+months_with_nans = list(set(nan_df.columns) - set(nan_df.dropna(axis="columns").columns))
+crime_data.drop(columns=months_with_nans, inplace=True)
+
+# major crime types are parents to minor categories
+major_counts_per_LSOA_per_month = crime_data.groupby(by=[JOINING_KEY, "Major Category"]).sum().astype('float').reset_index().rename(columns={"Major Category":"crime_category"})
+
+minor_counts_per_LSOA_per_month = crime_data.drop(columns="Major Category").rename(columns={"Minor Category":"crime_category"})
+
+# count crimes regardless of category for each LSOA (and check we have recordings on a monthly basis)
+total_counts_per_LSOA_per_month = crime_data.groupby(by=[JOINING_KEY]).sum().astype('float')
+assert sequential_months(set(total_counts_per_LSOA_per_month.columns)), "Unexpected number of months. Data may be missing for particular months"
+total_counts_per_LSOA_per_month["crime_category"] = "total_count"
+total_counts_per_LSOA_per_month = total_counts_per_LSOA_per_month.reset_index() # now joinable
+
+counts_per_LSOA_per_month = pd.concat([major_counts_per_LSOA_per_month, minor_counts_per_LSOA_per_month, total_counts_per_LSOA_per_month])
+
+# reduced previous count table to an overview aggregating across the months and calculate overall total counts, number of months and mean monthly crime count
+counts_per_LSOA = counts_per_LSOA_per_month.set_index([JOINING_KEY, "crime_category"]).apply(lambda x : x.sum(), axis=1).rename("crime_count").reset_index()
+counts_per_LSOA["n_months"] = counts_per_LSOA_per_month.set_index([JOINING_KEY, "crime_category"]).apply(lambda x : x.count(), axis=1).values
+counts_per_LSOA["mean_monthly_crime_count"] = counts_per_LSOA_per_month.set_index([JOINING_KEY, "crime_category"]).apply(lambda x : x.mean(), axis=1).values
+counts_per_LSOA["std_monthly_crime_count"] = counts_per_LSOA_per_month.set_index([JOINING_KEY, "crime_category"]).apply(lambda x : x.std(), axis=1).values
+```
+
+
+
+
+
+
+## Indices of deprivation & population density data
+
+To control for various indices of deprivation in the analysis, we load [data](https://data.london.gov.uk/dataset/indices-of-deprivation) indicating such measures provided by the Government (see 'IMDB2015' sheet in .xls). 
+In addition, to control for population density as a factor influencing crime rates, we also load demographic and related data, [Current LSOA boundaries post-2011](https://data.london.gov.uk/dataset/lsoa-atlas).
+
+
+
+
+
+
+```python
+depriv_indices = pd.read_csv("resources/ID 2015 for London exported.csv")
+
+# make sure to remove all indices 'directly' relating to crime
+depriv_indices.drop(columns=["Crime Score", "Crime Rank (where 1 is most deprived)", "Crime Decile (where 1 is most deprived 10% of LSOAs)"], inplace=True)
+
```

Comment:
We also only want to merge the "Score" columns of the other categories, so you might want to drop the rank and decile columns in general too. Far as I gather these are still getting pulled in?